core/src/main/java/org/openedx/core/domain/model/Progress.kt

```diff
@@ -8,6 +8,13 @@ data class Progress(
     val assignmentsCompleted: Int,
     val totalAssignmentsCount: Int,
 ) : Parcelable {
+
+    fun getProgress(): Float = try {
+        assignmentsCompleted.toFloat() / totalAssignmentsCount.toFloat()
+    } catch (_: ArithmeticException) {
+        0f
+    }
```

Comment:
What about replacing it with a computed property, let's say `value`, so we can call progress like this `progress.value` instead of `progress.getProgress()`?